lib/srv/discovery/fetchers/db/aws.go

```diff
@@ -17,11 +17,122 @@ limitations under the License.
 package db
 
 import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
 	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/srv/discovery/common"
 )
 
+// awsFetcherPlugin defines an interface that provides database type specific
+// functions for use by the common AWS database fetcher.
+type awsFetcherPlugin interface {
```

Comment:
Plugin is quite misleading what about awsFetcher where the object be be renamed to `awsFetcher` ?

Reply:
This is consistent with https://github.com/gravitational/teleport/blob/88d0c5c457f07a8af23369d181d59ea4e41f826d/lib/srv/discovery/fetchers/db/azure.go#L129

I thought the name made sense, because the "plugin" only provides partial behavior for a fetcher